src/ray/raylet/node_manager.h

```diff
@@ -122,6 +122,12 @@ struct NodeManagerConfig {
   absl::flat_hash_map<std::string, std::string> labels;
 };
 
+enum RayletShutdownState : std::uint8_t {
+  ALIVE,
+  SHUTDOWN_QUEUED,
+  SHUTTING_DOWN,
```

Comment:
the semantics and difference between these is nontrivial, please add a comment here describing the valid transitions